wagtail/admin/active_filters.py

```diff
@@ -0,0 +1,208 @@
+        query_dict.setlist(
+            param, [v for v in query_dict.getlist(param) if v != str(value)]
+        )
+        return self.base_url + "?" + query_dict.urlencode()
+
+    def get_active_filters(self):
+        """
+        Return a list of `ActiveFilter` objects representing the filter values
+        that are currently in force on the listing queryset.
+        """
+        yield ActiveFilter(
+            self.bound_field.auto_id,
+            self.filter.label,
+            str(self.value),
+            self.get_url_without_filter_param(self.bound_field.name),
+        )
+
+
+class ChoiceFilterAdapter(BaseFilterAdapter):
+    def get_active_filters(self):
+        choices = flatten_choices(self.filter.field.choices)
+        yield (
+            ActiveFilter(
+                self.bound_field.auto_id,
+                self.filter.label,
+                choices.get(str(self.value), str(self.value)),
+                self.get_url_without_filter_param(self.bound_field.name),
+            )
+        )
+
+
+class MultipleChoiceFilterAdapter(BaseFilterAdapter):
+    def get_active_filters(self):
+        choices = flatten_choices(self.filter.field.choices)
+        yield from (
+            ActiveFilter(
+                self.bound_field.auto_id,
+                self.filter.label,
+                choices.get(str(item), str(item)),
+                self.get_url_without_filter_param_value(self.bound_field.name, item),
+            )
+            for item in self.value
+        )
+
+
+class ModelChoiceFilterAdapter(BaseFilterAdapter):
+    def get_active_filters(self):
+        field: ModelChoiceField = self.filter.field
+        yield (
+            ActiveFilter(
+                self.bound_field.auto_id,
+                self.filter.label,
+                field.label_from_instance(self.value),
+                self.get_url_without_filter_param(self.bound_field.name),
+            )
+        )
+
+
+class ModelMultipleChoiceFilterAdapter(BaseFilterAdapter):
+    def get_active_filters(self):
+        field: ModelChoiceField = self.filter.field
+        yield from (
+            ActiveFilter(
+                self.bound_field.auto_id,
+                self.filter.label,
+                field.label_from_instance(item),
+                self.get_url_without_filter_param_value(self.bound_field.name, item.pk),
+            )
+            for item in self.value
+        )
+
+
+class RangeFilterAdapter(BaseFilterAdapter):
+    def format_value(self, value):
+        return str(value) if value is not None else ""
+
+    def get_active_filters(self):
+        start_date_display = self.format_value(self.value.start)
+        end_date_display = self.format_value(self.value.stop)
+        widget = self.filter.field.widget
+        yield (
+            ActiveFilter(
+                self.bound_field.auto_id,
+                self.filter.label,
+                f"{start_date_display} - {end_date_display}",
```

Comment:
I think you mixed Range and DateRange.
```suggestion
        start_value_display = self.format_value(self.value.start)
        end_value_display = self.format_value(self.value.stop)
        widget = self.filter.field.widget
        yield (
            ActiveFilter(
                self.bound_field.auto_id,
                self.filter.label,
                f"{start_value_display} - {end_value_display}",
```

Another thing I want to mention is the fact that one of the two range values (be them dates or anything) would result in a display like `"Number range: - 1000"` (missing start value) or `"Number range: 42 - "` (missing end value).

I would suggest to provide a `N/A` default to `format_value()` so that it's clear which range filtering values are set. An alternative would be to display either `"Number range: from {start_value}"` or `"Number range: until {end_value}"` depending on the presence.